contingency-api/src/test/java/eu/itesla_project/contingency/tasks/GeneratorTrippingTest.java

```diff
@@ -42,4 +49,22 @@ public void unknownGeneratorTrippingTest() {
         GeneratorTripping tripping = new GeneratorTripping("generator");
         tripping.modify(network, null);
     }
+
+    @Test
+    public void fictitiousSwitchTest() {
+        Network network = FictitiousSwitchFactory.create();
+        network.getSwitch("BT").setFictitious(true);
+
+        GeneratorTripping tripping = new GeneratorTripping("CD");
+
+        Set<Switch> switchesToOpen = new HashSet<>();
+        Set<Terminal> terminalsToDisconnect = new HashSet<>();
+        tripping.traverse(network, null, switchesToOpen, terminalsToDisconnect);
+        assertEquals(Collections.singleton("BJ"), switchesToOpen.stream().map(Switch::getId).collect(Collectors.toSet()));
+        assertEquals(Collections.emptySet(), terminalsToDisconnect);
+
+        tripping.modify(network, null);
+
+        assertTrue(network.getSwitch("BJ").isOpen());
```

Comment:
Maybe it would be more robust to assert not only open switches but the position of all of them?